pkg/vdso/vdso.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vdso
+
+import (
+	"debug/elf"
+	"fmt"
+
+	"github.com/google/pprof/profile"
+	"go.uber.org/multierr"
+
+	"github.com/parca-dev/parca-agent/pkg/metadata"
+	"github.com/parca-dev/parca-agent/pkg/objectfile"
+	"github.com/parca-dev/parca/pkg/symbol/symbolsearcher"
+)
+
+type Cache struct {
+	searcher symbolsearcher.Searcher
+	f        string
+}
+
+func NewCache() (*Cache, error) {
+	kernelVersion, err := metadata.KernelRelease()
+	if err != nil {
+		return nil, err
+	}
+	var (
+		elfFile *elf.File
+		merr    error
+		f       string
+	)
+	// find a file is enough
+	for _, vdso := range []string{"vdso.so", "vdso64.so"} {
+		f = fmt.Sprintf("/usr/lib/modules/%s/vdso/%s", kernelVersion, vdso)
+		elfFile, err = elf.Open(f)
```

Comment:
We should add an error handling code in case we fail to open the elf file. Also, the elf file will need to be closed after the usage.